### PR TITLE
Cortex_M backend: Fuse clamp + hardswish decompostion

### PR DESCRIPTION
**Pull request by AdrianLundell**
Adds quantization and fusion of clamp.

This is in turn used to decompose the hardswish operator in two passes, one clamping the dynamic range before quantization, and one decomposing the reminder of the operation into a maximum and mul op.

The tests in this patch exposes an issue in the runtime dim_order check as it cannot differ between channels_last/channels_first for tensors with C=1 or H=W=1. Therefore the check is removed and added as a TBD instead.

Additionally fixes per_tensor quantization for conv2d.

### CODE REVIEW

File: backends/cortex_m/ops/cortex_m_ops_common.h [hunk 1]
```diff
@@ -69,11 +69,7 @@ inline void validate_cmsis_nn_tensor_requirements(
         "Output must have the same sizes as inputs");
   }
 
-  // Dim order consistency
```
Comment (psiddh) [inline backends/cortex_m/ops/cortex_m_ops_common.h:72]: Removing this is bit risky.  I have debugged this issue further, and created a follow up task:
https://github.com/pytorch/executorch/issues/16032   (Pls update the comment with this issue no. just so that it doesn't fall through the cracks)

I will take the AI to fix the util function  + add test cases to test these edge cases and then add this check back

File: backends/cortex_m/passes/clamp_hardswish_pass.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import torch
+
+from executorch.exir.dialects.edge._ops import EdgeOpOverload
+from executorch.exir.pass_base import ExportPass, NodeMetadata, ProxyValue
+from torch.fx.node import Argument
+
+
+class ClampHardswishPass(ExportPass):
+    """
```
Comment (psiddh) [inline backends/cortex_m/passes/clamp_hardswish_pass.py:16]: +1 to the comment, this is cool

Are there test cases demonstrating the improved accuracy?  Also wondering if this paradigm could be applicable to other activation functions as well.

if so (not part of this PR, but in some future PR) consider extending this to a generic ClampActivationPass that works with configurable activation types and ranges.

File: backends/cortex_m/passes/decompose_hardswish_pass.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    """
+    Decomposes hardswish like
+
+        hardswish(x) = x * (clamp(x, -3, 3) + 3)/6
+
+    where the add and division is implemented by modifying the quantization parameters similar
+    to hardsigmoid in the activation_fusion_pass. Note that this pass assumes
+    that the output range of the preceding op is already clamped to [-3, inf] during
+    quantization by the clamp_hardswish_pass, removing the need for the negative clamp.
+    """
```
Comment (psiddh) [inline backends/cortex_m/passes/decompose_hardswish_pass.py:34]: +1 to the comment, helps the reviewer :D

**Comment by psiddh** [comment]
lgtm overall, although there is 1 major concern regd. dim order removal. But we have a clear action plan to address it properly and add it back (see https://github.com/pytorch/executorch/issues/16032 for more details)

**Review by psiddh — APPROVED** [review]

File: backends/cortex_m/ops/cortex_m_ops_common.h (hunk 1 above)
Comment (AdrianLundell) [reply backends/cortex_m/ops/cortex_m_ops_common.h:72]: Thanks!
